Add Benchmark 2.0 (#7888)
opensearch-project/documentation-website: _benchmark/reference/commands/run.md

---
layout: default
title: run
nav_order: 65
parent: Command reference
grand_parent: OpenSearch Benchmark Reference
redirect_from:
  - /benchmark/commands/run/
  - /benchmark/commands/execute-test/
---

<!-- vale off -->
# run
<!-- vale on -->

Whether you're using the included [OpenSearch Benchmark workloads](https://github.com/opensearch-project/opensearch-benchmark-workloads) or a [custom workload]({{site.url}}{{site.baseurl}}/benchmark/creating-custom-workloads/), use the `run` command to gather data about the performance of your OpenSearch cluster according to the selected workload. 

## Usage

> The following example tests a run using the `geonames` workload in test mode: 

Reviewer: Might be more clear to use: The following example performs a quick test using the `geonames` workload in test mode.

## Options

Use the following options to customize the `run` command for your use case. Options in this section are categorized by their use case. 

## General settings

> The following options customize each run and affects how run results appear: 

Reviewer: I think we can reword this to be --> ...customize each test run and affects how results appear

> The following options customize each run and affects how run results appear: 

Author: ```suggestion
The following options customize each test run and affects how run results appear: 
```

### Workload

> The following options determine which workload is used during the run:

Reviewer: I think `during the test` sounds more natural

> The following options determine which workload is used during the run:

Author: ```suggestion
The following options determine which workload is used during the test:
```

- `--workload-repository`: Defines the repository from which OpenSearch Benchmark loads workloads. 
- `--workload-path`: Defines the path to a downloaded or custom workload.
- `--workload-revision`: Defines a specific revision from the workload source tree that OpenSearch Benchmark should use.
- `--workload`: Defines the workload to use based on the workload's name. You can find a list of preloaded workloads using `opensearch-benchmark list workloads`.

> ### Run procedures

Reviewer: Run Procedures -> Scenarios

> ### Run procedures

Author: ```suggestion
### Scenarios
```

> The following options define what run procedures the run uses and which operations are contained inside the procedure: 

Reviewer: run procedures --> scenarios
the run uses --> the test uses

> The following options define what run procedures the run uses and which operations are contained inside the procedure: 

Author: ```suggestion
The following options define what scenarios the test uses and which operations are contained inside the procedure: 
```

> - `--test-run-id`: Defines a unique ID for this run.

Reviewer: Should we simplify this to `test-id` instead of `test-run-id`? @gkamat We haven't made this change yet so it's possible to switch to `test-id` to keep it simple...
